test/optim/test_optim.py

```diff
@@ -828,10 +828,11 @@ def _test_foreach_memory(self, optimizer_pairs_with_flags):
                     # bias_correction, mus, and mu_nexts
                     nintermediates = 5
 
-            elif optimizer_constructor.__name__ in ["NAdam", "Adagrad", "RMSprop"]:
+            elif optimizer_constructor.__name__ in ["NAdam", "Adagrad", "RMSprop", "Adamax"]:
                 # NAdam uses two intermediates at the same time (grads & exp_avg_sq_sqrt)
                 # Adagrad uses std and grads at the same time
                 # RMSprop uses avg and grads
+                # Adamax uses numerator and grads (maximize=True/weight_decay!=0)
```

Comment:
Ideally, we could keep intermediates lower--is this avoidable, do you think?

Reply:
Hmm, in our case, if we move the tensor `steps` into the `foreach_addcdiv_`, we trigger a type mismatch on dispatch. 

This then forces me to define an intermediate `numerator` applying the alpha value prior to `foreach_addcdiv_` rather than passing as an arg.

Basically, `foreach_addcdiv_` wants a list of scalars as its 4th arg. Ideally,  we can have best of both worlds - eager mode gets the CPU tensors, inductor mode gets the GPU tensor...

Let me reproduce the exact error.